src/vuex/modules/session.js

```diff
@@ -1,272 +1,264 @@
+        USER_PREFERENCES_KEY,
+        JSON.stringify({
+          cookiesAccepted: state.cookiesAccepted,
+          errorCollection: state.errorCollection,
+          analyticsCollection: state.analyticsCollection,
+          preferredCurrency: state.preferredCurrency
+        })
+      )
+    },
+    setErrorCollection({ state, dispatch }, enabled) {
+      // don't track in development
+      if (state.developmentMode) return
 
-        state.errorCollection = enabled
-        dispatch(`storeLocalPreferences`)
-      },
-      setAnalyticsCollection({ state, dispatch }, enabled) {
-        // don't track in development
-        if (state.developmentMode) return
+      state.errorCollection = enabled
+      dispatch(`storeLocalPreferences`)
+    },
+    setAnalyticsCollection({ state, dispatch }, enabled) {
+      // don't track in development
+      if (state.developmentMode) return
 
-        state.analyticsCollection = enabled
-        dispatch(`storeLocalPreferences`)
+      state.analyticsCollection = enabled
+      dispatch(`storeLocalPreferences`)
 
-        if (state.analyticsCollection) {
-          state.externals.deanonymize()
-          console.log(`Analytics collection has been enabled`)
-        } else {
-          state.externals.anonymize()
-          console.log(`Analytics collection has been disabled`)
-        }
-      },
-      setPreferredCurrency({ state, dispatch }, currency) {
-        state.preferredCurrency = currency
-        dispatch(`storeLocalPreferences`)
-      },
-      /* istanbul ignore next */
-      async checkAddressRole({ commit }, { address }) {
-        const api = await getAPI()
-        const bonded = await api.query.staking.bonded(address)
-        if (!bonded) {
-          // Set address role (stash | controller), useful for Polkadot networks so we can limit actions based on it
-          commit(`setUserAddressRole`, {
-            addressRole: `controller`
-          })
-          return
-        }
-        commit(`setUserAddressRole`, {
-          addressRole: `stash`
-        })
+      if (state.analyticsCollection) {
+        state.externals.deanonymize()
+        console.log(`Analytics collection has been enabled`)
+      } else {
+        state.externals.anonymize()
+        console.log(`Analytics collection has been disabled`)
       }
+    },
+    setPreferredCurrency({ state, dispatch }, currency) {
+      state.preferredCurrency = currency
+      dispatch(`storeLocalPreferences`)
+    },
+    /* istanbul ignore next */
+    async checkAddressRole({ commit }, { address, currentNetwork }) {
+      const { data } = await apollo.query({
+        query: AddressRole,
+        variables: { networkId: currentNetwork.id, address: address },
+        fetchPolicy: "network-only"
+      })
+      commit(`setUserAddressRole`, data.accountRole)
     }
+  }
 
-    return {
-      state,
-      mutations,
-      actions
-    }
+  return {
+    state,
+    mutations,
+    actions
   }
+}
```

Comment:
We really do need to fix the linting thing this file here is unreadable. Cannot really see the changes made 🤷